LethalAPI.Core/Attributes/LethalPluginAttribute.cs

```diff
@@ -0,0 +1,77 @@
+﻿// -----------------------------------------------------------------------
+// <copyright file="LethalPluginAttribute.cs" company="LethalAPI Modding Community">
+// Copyright (c) LethalAPI Modding Community. All rights reserved.
+// Licensed under the GPL-3.0 license.
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace LethalAPI.Core.Attributes;
+
+using System;
+
+/// <summary>
+/// Represents a plugin defined via attributes.
+/// </summary>
+/// <remarks>
+/// A class can also just inherit the IPlugin&lt;IConfig&gt; interface or Plugin&lt;IConfig&gt; class instead of using this.
+/// <code>
+/// class MyPlugin : Plugin&lt;MyConfig&gt;
+/// {
+/// }
+/// class MyConfig : IConfig
+/// {
+/// }
+/// </code>
+/// </remarks>
+[AttributeUsage(AttributeTargets.Class, Inherited = true, AllowMultiple = false)]
+public sealed class LethalPluginAttribute : Attribute
+{
+    /// <summary>
+    /// Initializes a new instance of the <see cref="LethalPluginAttribute"/> class.
+    /// </summary>
+    /// <param name="name">The name of the plugin.</param>
+    /// <param name="description">A description of the plugin.</param>
+    /// <param name="author">The author or authors of the plugin.</param>
+    /// <param name="version">The version of the plugin. You can also use the assembly versioning system and leave this null.</param>
+    public LethalPluginAttribute(string name, string description, string author, string version)
+    {
+        this.Name = name;
+        this.Description = description;
+        this.Author = author;
+        this.Version = Version.Parse(version);
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="LethalPluginAttribute"/> class.
+    /// </summary>
+    /// <param name="name">The name of the plugin.</param>
+    /// <param name="description">A description of the plugin.</param>
+    /// <param name="author">The author or authors of the plugin.</param>
+    public LethalPluginAttribute(string name, string description, string author)
```

Comment:
The plugin does an assembly version override check, as some people like to use the project settings to define general information about it. It's a good discussion point.

Reply:
> Should we _allow_ null versions? I intend on using these to compare versions for dependency checking and such. I think it's probably bad practice?

If a version is omitted, by default it should be 1.0.0 but if the author update it and didn't change the version so it might be an issue. So it is a bad practice to do that.

Reply:
I've updated the default to 1.0.0. It will check for assembly version first and use plugin version as a backup.